casts/modalNavigator/src/ModalNavigator/createModalNavigator.tsx

```diff
@@ -43,165 +33,44 @@ const ModalScreen = ({
     }, [name, descriptor.navigation]);
 
     return (
-        <NestedInModalContext.Provider value={hide}>
-            {descriptor.render()}
-        </NestedInModalContext.Provider>
+        <UIModalSheet visible={route.visible} onClose={hide} maxMobileWidth={maxMobileWidth}>
+            <NestedInModalContext.Provider value={hide}>
+                {descriptor.render()}
+            </NestedInModalContext.Provider>
+        </UIModalSheet>
     );
-};
-export class ModalController extends React.Component<ModalControllerProps> {
-    static show(name: string, params?: Record<string, unknown>) {
-        if (ModalController.instance) {
-            ModalController.instance.show(name, params);
-        }
-    }
-
-    static hide(name: string) {
-        if (ModalController.instance) {
-            ModalController.instance.hide(name);
-        }
-    }
-
-    static hideAll() {
-        if (ModalController.instance) {
-            ModalController.instance.hideAll();
-        }
-    }
-
-    componentDidMount() {
-        ModalController.instance = this;
-    }
-
-    componentWillUnmount() {
-        ModalController.instance = null;
-    }
-
-    static instance: ModalController | null = null;
-
-    show(name: string, params?: Record<string, unknown>) {
-        Keyboard.dismiss();
-        this.props.navigation.dispatch(ModalActions.show(name, params));
-    }
-
-    hide(name: string) {
-        Keyboard.dismiss();
-        this.props.navigation.dispatch(ModalActions.hide(name));
-    }
-
-    hideAll() {
-        Keyboard.dismiss();
-        this.props.navigation.dispatch(ModalActions.hideAll());
-    }
-
-    render() {
-        return this.props.state.routes.map<React.ReactNode>(route => {
-            const descriptor = this.props.descriptors[route.key];
-
-            if (descriptor == null) {
-                return null;
-            }
-
-            return <ModalScreen key={route.key} route={route} descriptor={descriptor} />;
-        });
-    }
 }
 
-type ModalSceneInstance = {
-    show: <Params = { [key: string]: any }>(params: Params) => void | Promise<void>;
-    hide: () => void;
-};
-
-type ModalSceneWrapperProps<ParamList extends ParamListBase = { params: { visible: boolean } }> = {
-    ref: React.Ref<ModalSceneInstance>;
-    navigation: NavigationProp<ParamList>;
-    route: RouteProp<ParamList, keyof ParamList>;
-};
-
-type ModalSceneWrapperState = {
-    visible: boolean;
-    needToHide: boolean;
-    needToShow: boolean;
-};
-
-export const withModalSceneWrapper = (
```

Comment:
I suggest to keep it until UIModalController is fully removed. Otherwise the migration process will turn into a nightmare... 🤪

Reply:
I'm fully prepared for this nightmare 💪Let's drop the legacy! :D